### ISSUE TITLE
No internet on emulator

### ISSUE THREAD

**Issue by JoshLudahl** [ReactiveCircus/android-emulator-runner#348]
When running tests, it detects the emulators have no internet connection. Is this disabled by default or is there a work-around?

Pixel 5 API 31. Tried a few devices.

**Comment by TobiGr** [comment]
Same happens with my tests. They fail regularly because of this.

**Comment by groot007** [comment]
same for me
github runner - macOs latest

I've tried to add DNS to host 
`etworksetup -setdnsservers Ethernet 8.8.8.8`

also tried to add 
`emulator-options: -dns-server 8.8.8.8`

but nothing helped

my configuration - macOs latest
`- name: Check connection on emulator
        uses: reactivecircus/android-emulator-runner@v2
        with:
          api-level: 31
          arch: x86_64
          profile: Nexus 6
          avd-name: ${{ steps.device.outputs.AVD_NAME }}
          force-avd-creation: false
          disable-animations: true
          emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
          script: adb shell ping -c 4 8.8.8.8`

**Comment by ShadyH20** [comment]
Same for me. Have you found a solution yet?

**Comment by JoshLudahl** [comment]
No. I ended up disabling the CI and just run the tests locally.
Unfortunately, it's not useful at this point.

**Comment by gcohen-dev** [comment]
This is happening on macOS or Ubuntu, any workaround?

**Comment by groot007** [comment]
I've tried to set DNS on MacOS runner and seems like now it's working

`networksetup -setdnsservers Ethernet 8.8.8.8 8.8.4.4`

and also set DNS in emulator options
`emulator-options: -no-snapshot-save -no-snapshot-load -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none **-dns-server 8.8.8.8**`

**Comment by theolm** [comment]
Same here. Any suggestions on how to make this work?

**Comment by rogervdf** [comment]
We solved this one, apparently, after a few days of tinkering.

It's related to https://github.com/ReactiveCircus/android-emulator-runner/issues/211. Also somehow to how xen virtualisation networking and virtual adapters work, perhaps.

On github we also noticed that the actual [runner](https://github.com/actions/runner) virtual machines being deployed somehow have influence on the xen wifi adapter connectivity. One run they behave, the next they don't, depending on which machine is assigned. At a minimum the logs are not consistent.

TLDR
If you run the emulator with:
`emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -dns-server 8.8.8.8 -netfast -no-metrics`
then connectivity is up. probably only the `dns-server` bit is required

**Comment by marcobeltempo** [comment]
I'm also experiencing this on `ubuntu-latest`. We've tried passing the `-dns-server` flag and setting the default ip route with no luck.

```yml
uses: reactivecircus/android-emulator-runner@v2
      with:
        api-level: 33
        arch: x86_64
        avd-name: ${{ inputs.device-profile }}
        cores: 4
        disable-animations: true
        disable-linux-hw-accel: auto
        force-avd-creation: true
        target: 'google_apis'
        emulator-options: -no-window -gpu swiftshader_indirect -no-snapshot -noaudio -no-boot-anim -writable-system -dns-server 8.8.8.8 -netfast
        script: |
          $ANDROID_HOME/platform-tools/adb shell ping -c 4 google.com
          killall -INT crashpad_handler

```

**Ping google.com**
```
PING google.com (172.217.12.110) 56(84) bytes of data.
From 10.0.2.2: icmp_seq=1 Destination Host Unreachable
From 10.0.2.2: icmp_seq=2 Destination Host Unreachable
From 10.0.2.2: icmp_seq=3 Destination Host Unreachable

--- google.com ping statistics ---
4 packets transmitted, 0 received, +3 errors, 100% packet loss, time 3060ms
```

**Comment by aphrysof** [comment]
@marcobeltempo  did you get to fix this issue?

**Comment by JoshLudahl** [comment]
I haven't tried again.

**Comment by marcobeltempo** [comment]
> [@marcobeltempo](https://github.com/marcobeltempo) did you get to fix this issue?

@aphrysof I haven't managed to get this working.
